### PR TITLE
Add my_connection_lost event and some fixes

### PR DESCRIPTION
**Pull request by a11cf0**
* Add a few missing string conversions for Linux commpatibility.
* Prevent automatically joining the root channel if a negative channel id is passed when connecting to a server.

### CODE REVIEW

File: teamtalk/instance.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def connect(self) -> bool:
             return False
         result, msg = _waitForEvent(self.super, sdk.ClientEvent.CLIENTEVENT_CON_SUCCESS)
         if not result:
+            self.bot.dispatch("my_connection_lost", self.server)
```
Comment (JessicaTegner) [inline teamtalk/instance.py:75]: This is not really how the event should be used. I'm thinking that, if we don't have a connection to begin with, we can't lose it. Aka, we can listen for it in the process message call and dispatch it that way, but shouldn't dispatch it during connection.

File: teamtalk/instance.py [hunk 2]
```diff
@@ -841,6 +842,9 @@ async def _process_events(self) -> None:  # noqa: C901
         if event == sdk.ClientEvent.CLIENTEVENT_CMD_MYSELF_KICKED:
             self.bot.dispatch("my_kicked_from_channel", TeamTalkChannel(self, msg.nSource))
             return
+        if event == sdk.ClientEvent.CLIENTEVENT_CON_LOST or event == sdk.ClientEvent.CLIENTEVENT_CON_FAILED:
+            self.bot.dispatch("my_connection_lost", self.server)
```
Comment (JessicaTegner) [inline teamtalk/instance.py:846]: So yeah basically like this, compared to what I said above

**Comment by kirill-jjj** [comment]
@JessicaTegner I know that this code is not beautiful, but we need it very much for our bot, which even when restarting the server must necessarily enter back to it. If possible, can you embed this code into the library? You can change it, I guess, but the main thing is to have this possibility.

**Comment by JessicaTegner** [comment]
@kirill-jjj  look above for comments I've made to the code, once those are addressed, I would be happy to merge it into main no problem :)

**Pull request closed without merging**